### ISSUE TITLE
How to get all nodes within a same layer?

### ISSUE THREAD

**Issue by ganshaoduo** [Microsoft/CNTK#2477]
Hi guys,

I think CNTK manages weights and gradients by node. But I want to get weights and gradients by layer in neural network, like ResNet.

So I wonder if I can get all nodes within a same layer? Or there are other ways to get weights and gradients by layer?

Thanks.

**Comment by cha-zhang** [comment]
When you train the model, did you name the layers? 

In general: 
https://docs.microsoft.com/en-us/cognitive-toolkit/How-do-I-Read-Things-in-Python

**Comment by ganshaoduo** [comment]
Hi,

I just used the default Brainscript Resnet20 model in CNTK examples. Also, I want to get weights and gradients of each iteration by layer during the training process. I know how to get this by node, but do not know how to get this by layer. 

Thanks.

**Comment by cha-zhang** [comment]
You can load model and call find_by_name() to access each layer.

**Comment by ganshaoduo** [comment]
Thanks, but you mean load the model during the training?

**Comment by cha-zhang** [comment]
If during training, you should have access to the model and layers directly. If you have a dense layer: 
```
z=Dense(100)(x)
```
z.parameters should give you access to the weights and bias.

**Comment by ganshaoduo** [comment]
OK, but if it's possible to get weights per layer in SGD.cpp ?

**Comment by cha-zhang** [comment]
Do you want to add some special regularization to the parameters? You are using C++? 

CNTK adopts a similar approach as Theano, where different regularizations can be applied to different layers by having different learners.
